mono/utils/mono-threads-coop.c

```diff
@@ -197,6 +198,53 @@ copy_stack_data (MonoThreadInfo *info, MonoStackData *stackdata_begin)
 	state->gc_stackdata_size = stackdata_size;
 }
 
+#ifdef _MSC_VER
+typedef void (*CopyStackDataFunc)(MonoThreadInfo *, MonoStackData *, gconstpointer, gconstpointer);
+
+#ifdef TARGET_AMD64
+typedef struct {
+	__m128d fregs [10];
+	host_mgreg_t gregs [8];
+} MonoBuiltinUnwindInfo;
+
+// Defined in win64.asm
+G_EXTERN_C void
+copy_stack_data_internal_win32_wrapper (MonoThreadInfo *, MonoStackData *, MonoBuiltinUnwindInfo *, CopyStackDataFunc);
+#else
+typedef struct {
+	host_mgreg_t gregs [4];
+} MonoBuiltinUnwindInfo;
+
+__declspec(naked) void
+copy_stack_data_internal_win32_wrapper (MonoThreadInfo *info, MonoStackData *stackdata_begin, MonoBuiltinUnwindInfo *unwind_info_data, CopyStackDataFunc func)
+{
+	__asm {
+		mov edx, dword ptr [esp + 0Ch]
+		mov dword ptr [edx + 00h], ebx
+		mov dword ptr [edx + 04h], esi
+		mov dword ptr [edx + 08h], edi
+		mov dword ptr [edx + 0Ch], ebp
+
+		mov edx, dword ptr [esp + 10h]
+		jmp edx
```

Comment:
Maybe a comment, that func takes same number of params as copy_stack_data_internal_win32_wrapper.
Or maybe it is blindingly obviously -- you wouldn't tailcall otherwise.

Maybe just add a comment to the last line "tailcall". but again, maybe blindingly obvious.
And there sure is a nice set of simplifications  -- no frame setup or teardown.